### PR TITLE
chore: enable branch protection rule and other configurations on digital-product-pass main branch

### PR DESCRIPTION
**Pull request by saudkhan116**
## Description

- Added branch protection main branch on [digital-product-pass](https://github.com/eclipse-tractusx/digital-product-pass) repository
- Enabled branch and gh-pages relevant configurations 
- Guided information from EF and Otterdog Team https://github.com/eclipse-tractusx/.eclipsefdn/issues/28

FYI @stephanbcbauer 

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

**Comment by netomi** [comment]
LGTM, once approved by a PL you can merge yourself, follow the instructions that will be added as comment.

**Review by matbmoser — APPROVED** [review]
Approved from my side, understood that is important to have rules in our repository that can preserve the security. At least one commiter review is required for every pull request to be merged to main. We also want to enabled discussions and the webpage for the documentation to be available.

**Review by netomi — APPROVED** [review]

**Pull request merged**
